junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/ClasspathAlignmentChecker.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher.core;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Comparator.comparing;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.junit.platform.commons.JUnitException;
+import org.junit.platform.commons.support.ReflectionSupport;
+import org.junit.platform.commons.util.ClassLoaderUtils;
+
+/**
+ * @since 1.12
+ */
+class ClasspathAlignmentChecker {
+
+	// VisibleForTesting
+	static final List<String> WELL_KNOWN_PACKAGES = unmodifiableList(Arrays.asList( //
+		"org.junit.jupiter.api", //
+		"org.junit.jupiter.engine", //
+		"org.junit.jupiter.migrationsupport", //
+		"org.junit.jupiter.params", //
+		"org.junit.platform.commons", //
+		"org.junit.platform.console", //
+		"org.junit.platform.engine", //
+		"org.junit.platform.jfr", //
+		"org.junit.platform.launcher", //
+		"org.junit.platform.reporting", //
+		"org.junit.platform.runner", //
+		"org.junit.platform.suite.api", //
+		"org.junit.platform.suite.commons", //
+		"org.junit.platform.suite.engine", //
+		"org.junit.platform.testkit", //
+		"org.junit.vintage.engine" //
+	));
+
+	static Optional<JUnitException> check(LinkageError error) {
+		ClassLoader classLoader = ClassLoaderUtils.getClassLoader(ClasspathAlignmentChecker.class);
+		Function<String, Package> packageLookup = name -> ReflectionSupport.findMethod(ClassLoader.class,
+			"getDefinedPackage", String.class) //
+				.map(m -> (Package) ReflectionSupport.invokeMethod(m, classLoader, name)) //
+				.orElseGet(() -> getPackage(name));
```

Comment:
This only works for packages for which classes have already been loaded by the class loader and thus is likely to contain the conflicting JARs.